File: test_shopee_last_try/2.py
import pandas as pd

df1 = pd.read_csv('Analisis excel/Youtube/5 soal/Customer.csv')
print(df1)
df1 = df1.groupby('City')['Customer ID'].unique().explode().reset_index()

df2 = pd.read_csv('Analisis excel/Youtube/5 soal/Orders.csv')
print(df2)
df = pd.merge(df1, df2, on='Customer ID', how='inner')
print(df.head())

hasil = df.groupby('City')['Order ID'].count().reset_index()
print(hasil)

File: test_shopee_last_try/2.r
library(dplyr)
library(tidyr)

customer <- read.csv("Analisis excel/Youtube/5 soal/Customer.csv")
customer <- customer %>% group_by(City) %>% reframe(Customer.ID = unique(Customer.ID))
# customer <- customer %>% group_by(City) %>% summarise(Customer_ID = list(unique(Customer.ID))) %>% unnest(cols = Customer_ID)
print(head(customer))

order <- read.csv("Analisis excel/Youtube/5 soal/Orders.csv")
print(head(order))
cat('\n')


# Inner join customer and order data frames
hasil1 <- inner_join(customer, order, by = "Customer.ID", relationship ="many-to-many")
print(head(hasil1))

cat("\n\n")

# Alternative approach using merge
hasil2 <- merge(customer, order, by = "Customer.ID", all = FALSE)
print(head(hasil2))

hasil_akhir <- hasil1 %>% group_by(City) %>% summarise(jumlah = n()) 
print(hasil_akhir)

hasil_akhir <- hasil2 %>% group_by(City) %>% summarise(jumlah = n()) 
print(hasil_akhir)